_posts/2018-01-05-power-of-memory.md

```diff
@@ -0,0 +1,195 @@
+These are \\(CC, CD, DC\\) and \\(DD\\).
+
+A memory one strategy is denoted by the probabilities of cooperating after
+each of these states, \\(p = (p_1, p_2, p_3, p_4) \in R_{[0,1]} ^ 4\\).
+
+Press and Dyson found a way for a memory one opponent \\(p\\) to manipulate an
+opponent \\(q\\) and they called these "manipulative" strategies, **zero determinant** strategies.
+Moreover, Press and Dyson stated that in a two players interaction,
+a player playing a zero determinant strategy can outdo any longer memory strategy.
+Concluding that in the iterated prisoner's dilemma a longer memory size than 1
+is not advantageous.
+
+The purpose of my project is to show that memory one strategies have limitations.
+In order to achieve that I want to initially identify the best response
+\\(p^*=(p_1, p_2, p_3, p_4)\\) to a strategy \\(q\\). In essence answering the
+question:
+
+**What is the best memory one strategy against a given other memory one strategy?**
+
+<p align="center">
+  <img src="{{site.baseurl}}/assets/images/against_one_player.png" style='height: 20%; width: 30%; object-fit: contain'>
+</p>
+
+A match between two memory one players \\(p\\) and \\(q\\) can be modelled as a
+stochastic process, where the players move from state to  state. More specifically,
+it can be model by the use of a [Markov chain](https://en.wikipedia.org/wiki/Markov_chain),
+which is described by matrix \\(M\\).
+
+\\[M =
+\begin{bmatrix}
+    p_{1} q_{1} & p_{1} (- q_{1} + 1) & q_{1} (- p_{1} + 1) & (- p_{1} + 1) (- q_{1} + 1)
+    \\\
+    p_{2} q_{3} & p_{2} (- q_{3} + 1) & q_{3} (- p_{2} + 1) & (- p_{2} + 1) (- q_{3} + 1)
+    \\\
+    p_{3} q_{2} & p_{3} (- q_{2} + 1) & q_{2} (- p_{3} + 1) & (- p_{3} + 1) (- q_{2} + 1)
+    \\\
+    p_{4} q_{4} & p_{4} (- q_{4} + 1) & q_{4} (- p_{4} + 1) & (- p_{4} + 1) (- q_{4} + 1)
+    \\\
+\end{bmatrix}
+\\]
+
+Our players are assumed to move from each state until the system reaches a
+state steady. Thus in order to calculate the utility of our players we
+multiply the steady states of \\(M\\) with the payoffs of the game matrix.
+
+More specifically, the first result discussed here is that the utility of player
+\\(p\\) against an opponent \\(q\\), denoted as \\(u_q(p)\\), can be
+written in compact way:
+
+\\[u_q(p) = \frac{\frac{1}{2}p^TQ + c^Tp + a}
+            {\frac{1}{2}p^T\bar{Q} + \bar{c}^Tp + \bar{a}}\\]
+
+where \\(Q, \bar{Q}\\) are matrices of \\(4 \times 4\\), and \\(c, \bar{c}\\) are
+\\(4 \times 1\\) vectors defined with the transition probabilities of the
+opponent's transition probabilities \\(q_1, q_2, q_3, q_4\\).
+
+Thus in order to answer our question, `what is the best memory one strategy against
+a given other memory one strategy?`, we need to consider the following optimization
+problem:
+
+\\[ max_{q}: u_q(p) \\]
+\\[st: \\ p \in\mathbb{R}_{[0,1]}^{4}\\]
+
+Solving analytically a maximisation problem of \\(16\\) can be proven to be
```

Comment:
I don't know what you mean by `16` here?

Perhaps you mean: `Obtaining an analytical solution to this problem is not immediate.`